saltgui/static/scripts/routes/keys.js

```diff
@@ -6,23 +6,57 @@ class KeysRoute extends PageRoute {
     this.jobsLoaded = false;
 
     this._updateKeys = this._updateKeys.bind(this);
+    this._updateKeysFingerprint = this._updateKeysFingerprint.bind(this);
   }
 
   onShow() {
     const keys = this;
+    const p1 = keys.router.api.getKeys();
+    const p2 = keys.router.api.getKeysFingerprint();
+
+    Promise.all([p1, p2])
+      .then(function(data){
+        // process result of 1st promise
+        keys._updateKeys(data[0]);
+        // process result of 2nd promise
+        keys._updateKeysFingerprint(data[1]);
+      });
     return new Promise(function(resolve, reject) {
       keys.resolvePromise = resolve;
       if(keys.keysLoaded && keys.jobsLoaded) resolve();
-      keys.router.api.getMinions().then(keys._updateMinions);
-      keys.router.api.getKeys().then(keys._updateKeys);
       keys.router.api.getJobs().then(keys._updateJobs);
       keys.router.api.getJobsActive().then(keys._runningJobs);
     });
   }
 
+  _updateKeysFingerprint(data) {
+    const keys = data.return[0].data.return;
+
+    for(const property of Object.keys(keys)) {
+      if(property === "local") continue;
+      const hosts = keys[property];
+      for(const hostname of Object.keys(hosts)) {
+        const fingerprint = hosts[hostname];
+        // why this is sometimes null?
+        const element = document.querySelector("#" + hostname);
+        if(element) {
+          // remove td.os for accepted minions and add td.fingerprint
+          for (const item of element.children) {
+            if(item.className === "os") {
```

Comment:
better just remove all content from the `tr`. then add all needed fields. that is what happens on other pages

Reply:
yes but you must loop again over different minion statuses to build proper drop down menu which is waist of cpu cycles. If it was simple menu entry then probably yes but here I don't think it's beneficial, better loop over small children and remove os.

Reply:
@erwindon is this explanation enough?